File: src/lessons/welcome/bdr/extended/BDR2Entity.py
def getIndication():
	if isOverMessage():
		return readMessage()[0]
	else:
		return ' '
	
# BEGIN SOLUTION
def doR(): turnRight(); forward()
def doL(): turnLeft();  forward()
def doI(): turnBack();  forward()

danceOneStep = { 
  'R':doR,
  'L':doL,
  'I':doI,
  'A':forward(1),
  'B':forward(2),
  'C':forward(3),
  'D':forward(4),
  'E':forward(5),
  'F':forward(6),

  'Z':backward(1),
  'Y':backward(2),
  'X':backward(3),
  'W':backward(4),
  'V':backward(5),
  'U':backward(6),
}


while getIndication() != ' ':
	danceOneStep[getIndication()]()
# END TEMPLATE

File: src/lessons/welcome/bdr/extended/BDR2Entity.java
package lessons.welcome.bdr.extended;

import jlm.universe.bugglequest.SimpleBuggle;

public class BDR2Entity extends SimpleBuggle {
	public char getIndication() { 
		if (isOverMessage()) { 
			return readMessage().charAt(0); 
		} else { 
			return ' '; 
		} 
	}

	/* BEGIN TEMPLATE */
	boolean moreMusic = true;

	public void danceOneStep() {
		/* BEGIN SOLUTION */
		switch (getIndication()) {
		case 'R': turnRight(); forward(); break;
		case 'L': turnLeft();  forward(); break;
		case 'I': turnBack();  forward(); break;

		case 'A': forward(1); break;
		case 'B': forward(2); break;
		case 'C': forward(3); break;
		case 'D': forward(4); break;
		case 'E': forward(5); break;
		case 'F': forward(6); break;

		case 'Z': backward(1); break;
		case 'Y': backward(2); break;
		case 'X': backward(3); break;
		case 'W': backward(4); break;
		case 'V': backward(5); break;
		case 'U': backward(6); break;

		default: moreMusic = false;
		}
		/* END SOLUTION */
	}

	public void run() { 
		while (moreMusic)
			danceOneStep();
	}
	/* END TEMPLATE */
}
